pyrobosim/pyrobosim/core/yaml_utils.py

```diff
@@ -75,7 +76,12 @@ def create_world(self):
                 )
             else:
                 obj_data = None
-            self.world.set_metadata(locations=loc_data, objects=obj_data)
+
+            loc_data = [loc_data] if isinstance(loc_data, str) else loc_data or []
+            obj_data = [obj_data] if isinstance(obj_data, str) else obj_data or []
```

Comment:
lots of logic baked into here... is this to handle the metadata being a single string vs. a list of them?

Reply:
Yes.